src/coreclr/gc/gc.cpp

```diff
@@ -11541,6 +11532,477 @@ void gc_heap::rearrange_heap_segments(BOOL compacting)
 }
 #endif //!USE_REGIONS
 
+#if defined(USE_REGIONS)
+// trim down the list of free regions pointed at by free_link down to target_count, moving the extra ones to surplus_link
+static void remove_surplus_regions (region_free_list* free_list, region_free_list* surplus_list, size_t target_count)
+{
+    while (free_list->get_num_free_regions() > target_count)
+    {
+        // remove one region from the heap's free list
+        heap_segment* region = free_list->unlink_region_front();
+
+        // and put it on the surplus list
+        surplus_list->add_region_front (region);
+    }
+}
+
+// add regions from surplus_link to free_link, trying to reach target_count
```

Comment:
comment still calls it link instead of list

Reply:
Yup.